packages/salesforcedx-vscode-core/test/vscode-integration/util/cliConfiguration.test.ts

```diff
@@ -125,4 +131,45 @@ describe('SFDX CLI Configuration utility', () => {
       ]);
     });
   });
+
+  describe('ConfigAggregator integration tests', () => {
+    let localSfdxConfigFile: ConfigFile;
+    const dummyLocalDefaultUsername = 'test@local.com';
+
+    afterEach(() => {
+      // Clean up by removing the config file that was created in the setup for this test
+      localSfdxConfigFile.unlinkSync();
+    });
+
+    it.only('Should return the locally configured default username when it exists', async () => {
+      // Arrange: create a local config file with a local default username
+      const localSfdxDirPath = path.join(getRootWorkspacePath(), '.sfdx');
+      process.chdir(localSfdxDirPath);
+
+      class MyConfig extends ConfigFile {
+        public static getFileName(): string {
+          return 'sfdx-config.json';
+        }
+      }
+
+      localSfdxConfigFile = await MyConfig.create({
+        isGlobal: false
+      });
+      localSfdxConfigFile.set(
+        OrgConfigProperties.TARGET_ORG,
+        dummyLocalDefaultUsername
+      );
+      localSfdxConfigFile.writeSync(); // doesn't fire the file watcher logic to run
+      process.chdir('/'); // Change back to the default process.cwd
+
+      // Act
+      const localProjectDefaultUsernameOrAlias = await ConfigUtil.getDefaultUsernameOrAlias();
+
+      // Assert
+      // when the test initializes it creates a ConfigAggregator instance for the SFDX simple project, and put it in the config aggregator provider internal map. I have to get that provider and call reload in order to reset it before this test.
```

Comment:
Not sure if we can get this test to work.  I can set up the config file using the core lib types, but since a config aggregator instance is created for the project during the test initialization before the it block even runs, that global one is being returned for this project. In order to get the ConfigAggregator to be accurate for the project I would have to call reload, but I think that kind of defeats the point of this test... Open to any suggestions or ideas here.

Reply:
So is this identifying a current error condition we have with the flow of auth'ing an org for the first time? 
The steps seems like: 
1. VSCode opens and extensions load (no local config so global configAgg gets populated).
2. The localConfig gets set some how ( not sure if above flow is same as what we have when a user auths or not). 
3. We should ensure that all configAggs get reloaded to the local value.